extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/OpentelemetryExemplarSamplerProvider.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.micrometer.runtime.export;
+
+import java.util.Optional;
+
+import javax.enterprise.inject.Produces;
+
+import io.prometheus.client.exemplars.DefaultExemplarSampler;
+import io.prometheus.client.exemplars.ExemplarSampler;
+import io.prometheus.client.exemplars.tracer.common.SpanContextSupplier;
+import io.quarkus.opentelemetry.runtime.OpenTelemetryUtil;
+import io.quarkus.vertx.core.runtime.VertxMDC;
+
+public class OpentelemetryExemplarSamplerProvider {
+
+    @Produces
+    public Optional<ExemplarSampler> exemplarSampler() {
+        return Optional.of(new DefaultExemplarSampler(new SpanContextSupplier() {
+            @Override
+            public String getTraceId() {
+                return VertxMDC.INSTANCE.get(OpenTelemetryUtil.TRACE_ID);
```

Comment:
Is there any other place we can get this from? Can/should we store this information in the Vertx request context earlier? I forget if the metrics call is inside of the span or outside (I think inside?)

Reply:
I'll check that point.